src/internal_events/dnstap.rs

```diff
@@ -10,32 +10,42 @@ pub struct DnstapEventReceived {
 
 impl InternalEvent for DnstapEventReceived {
     fn emit_logs(&self) {
-        trace!(message = "Received line.", byte_size = %self.byte_size);
+        trace!(message = "Events received.", count = 1, byte_size = %self.byte_size);
     }
 
     fn emit_metrics(&self) {
-        counter!("processed_events_total", 1);
         counter!("component_received_events_total", 1);
+        counter!("component_received_event_bytes_total", self.byte_size as u64);
+        // deprecated
+        counter!("processed_events_total", 1);
         counter!("events_in_total", 1);
         counter!("processed_bytes_total", self.byte_size as u64);
     }
 }
 
 #[derive(Debug)]
-pub struct DnstapParseDataError<'a> {
+pub struct DnstapParseError<'a> {
     pub error: &'a str,
 }
 
-impl<'a> InternalEvent for DnstapParseDataError<'a> {
+impl<'a> InternalEvent for DnstapParseError<'a> {
     fn emit_logs(&self) {
         error!(
-            target = "dnstap event",
             message = "Error occurred while parsing dnstap data.",
             error = ?self.error,
-            internal_log_rate_secs = 10);
+            error_type = "parse_failed",
+            stage = "processing",
+            internal_log_rate_secs = 10,
+        );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
```

Comment:
Is this error discarded?

https://github.com/vectordotdev/vector/blob/8348b5d079a397da21a2903efaab5626ff4ce6bd/src/sources/dnstap/mod.rs#L197-L200 ... suggests it might be.

If it _is_, then `component_discarded_events_total` should be incremented too per https://github.com/vectordotdev/vector/blob/master/docs/specs/component.md#error.